### PR TITLE
kargs: Introduce usage of fs_utf8

### PR DESCRIPTION
**Pull request by cgwalters**
This avoids annoyances with constantly checking for utf-8.

### CODE REVIEW

File: lib/src/kargs.rs [hunk 1]
```diff
@@ -34,24 +35,27 @@ impl Config {
...
-        let name = name
-            .to_str()
-            .ok_or_else(|| anyhow::anyhow!("Invalid non-UTF8 filename: {name:?}"))?;
-        if !Config::filename_matches(name) {
-            continue;
-        }
-        let buf = d.read_to_string(name)?;
+    let mut entries = d
+        .entries()?
+        .try_fold(Vec::new(), |mut acc, ent| -> Result<_> {
```
Comment (cgwalters) [inline lib/src/kargs.rs:48]: This one can be replaced by `filenames_filtered` from https://github.com/coreos/cap-std-ext/pull/53

**Review by jmarrero — APPROVED** [review]
lgtm

**Review by vrothberg — APPROVED** [review]
LGTM

**Comment by cgwalters** [comment]
I'm going to redo this on top of https://github.com/coreos/cap-std-ext/pull/53 once it's released

**Comment by cgwalters** [comment]
> I'm going to redo this on top of https://github.com/coreos/cap-std-ext/pull/53 once it's released

Done :heavy_check_mark:

**Review by jmarrero — APPROVED** [review]
lgtm

**Pull request merged**